### PR TITLE
Feature/group description

### PR DESCRIPTION
**Pull request by rsommerfeld**
Chat info pages show the description for groups now. Nothing has changed for private chats.

### CODE REVIEW

File: packages/TelegramClient-Core.package/TCCChat.class/instance/isNotPrivate.st [hunk 1]
```diff
@@ -1,4 +0,0 @@
-initialization
-isNotPrivate
-
-	^ self isPrivate not
```
Comment (permler) [inline packages/TelegramClient-Core.package/TCCChat.class/instance/isNotPrivate.st:4]: Why did you delete this?

File: packages/TelegramClient-Core.package/TCCGroup.class/instance/isGroup.st [hunk 2]
```diff
@@ -0,0 +1,4 @@
+accessing
+isGroup
+
+	^ true
```
Comment (permler) [inline packages/TelegramClient-Core.package/TCCGroup.class/instance/isGroup.st:4]: We thought it was weird to have a method isGroup for elements of class Group but I guess it is fine for now.

File: packages/TelegramClient-UI.package/TCUChatInfoPage.class/instance/addDescriptionText.st [hunk 3]
```diff
@@ -0,0 +1,8 @@
+drawing
+addDescriptionText
+
+	self description addMorph: (TextMorph new
+		contents: self chat description asText;
+		wrapFlag: true;
+		hResizing: #spaceFill;
+		lock).
```
Comment (permler) [inline packages/TelegramClient-UI.package/TCUChatInfoPage.class/instance/addDescriptionText.st:8]: asText should not be necessary

File: packages/TelegramClient-Core.package/TCCGroup.class/instance/isGroup.st (hunk 2 above)
Comment (phkeese) [reply packages/TelegramClient-Core.package/TCCGroup.class/instance/isGroup.st:4]: > We thought it was weird to have a method isGroup for elements of class Group but I guess it is fine for now.

It is a subclass of ``TCCChat` which has an `isGroup` method

File: packages/TelegramClient-Core.package/TCCChat.class/class/newFromChatEvent..st [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
-		title: (chat at: 'title');
-		id: (chat at: 'id');
-		canSendMessages: ((chat at: 'permissions') at: 'can_send_messages');
-		type: ((chat at: 'type') at: '@type');
-		lastMessage: '';
+	| type chat |
+	type := ((anEvent at: 'chat') at: 'type') at: '@type'.
+	
+	((type = 'chatTypeSupergroup') or: (type = 'chatTypeBasicGroup'))
+		ifTrue: [chat := TCCGroup new]
```
Comment (phkeese) [inline packages/TelegramClient-Core.package/TCCChat.class/class/newFromChatEvent..st:8]: a bit weird to create a chat and then initialize it, why not use a constructor here? 
If true, return a TCCGroup from the block, if false return a `TCCChat`. Construct both in their respective blocks

File: packages/TelegramClient-Core.package/TCCChat.class/instance/initializeFromChatEvent..st [hunk 5]
```diff
@@ -0,0 +1,11 @@
+initialization
+initializeFromChatEvent: anEvent
```
Comment (phkeese) [inline packages/TelegramClient-Core.package/TCCChat.class/instance/initializeFromChatEvent..st:2]: see above

File: packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/addChat..st [hunk 6]
```diff
@@ -4,12 +4,12 @@ addChat: aChatEvent
...
-		self loadBasicGroupChatMemberCountFor: chat with: (((aChatEvent at: 'chat') at: 'type') at: 'basic_group_id').
+		self loadBasicGroupInfoFor: chat with: (((aChatEvent at: 'chat') at: 'type') at: 'basic_group_id').
 	].
 
 	self app chats
-		add: chat; 
-		notify.
\ No newline at end of file
+			add: chat; 
+			notify.
```
Comment (phkeese) [inline packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/addChat..st]: why indent like this? A single indentation should be just as readable, as the subject is on its own line

File: packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/loadBasicGroupInfoFor.with..st [hunk 7]
```diff
@@ -0,0 +1,13 @@
+events
+loadBasicGroupInfoFor: aChat with: aBasicGroupId
+
+	self client send: (TCCRequest
+		newWithType: 'getBasicGroupFullInfo'
+		from: {
+		'basic_group_id'->aBasicGroupId.
+		'@extra'->aBasicGroupId}).
```
Comment (phkeese) [inline packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/loadBasicGroupInfoFor.with..st:8]: indent once more to make more readable

File: packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/loadBasicGroupInfoFor.with..st [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+	
+	self app pendingRequests at: aBasicGroupId asString put: [:aBasicGroupFullInfoEvent |
```
Comment (phkeese) [inline packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/loadBasicGroupInfoFor.with..st:10]: please use `TCCCore>>registerCallback:with:`

File: packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/loadSuperGroupInfoFor.with..st [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+	self client send: (TCCRequest
+		newWithType: 'getSupergroupFullInfo'
+		from: {
+		'supergroup_id'->aSupergroupId.
+		'@extra'->aSupergroupId}).
+	
+	self app pendingRequests at: aSupergroupId asString put: [:aSupergroupFullInfoEvent |
+		aChat
+			memberCount: (aSupergroupFullInfoEvent at: 'member_count');
+			description: (aSupergroupFullInfoEvent at: 'description').].
```
Comment (phkeese) [inline packages/TelegramClient-Core.package/TCCChatsHandler.class/instance/loadSuperGroupInfoFor.with..st:13]: please use `TCCCore>>registerCallback:with:`

File: packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testDoesDisplayGroupDescription.st [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+	[self core chats anySatisfy: [:c | c isGroup]]
+		whileFalse: [6 seconds wait.].
+	groupChat := (self core chats detect: [:chat | chat isGroup]).
+	groupChat description: TCTMocks mockDescription.
+	
+	(self subject findByCriteria: [ :morph | (morph respondsTo: #chatID) and: [morph chatID = groupChat id]]) click.
+	titleBar := (self subject findByClass: TCUChatWindow) morphs collect: [:chatWindow | chatWindow titleBar].
+	(self subject createWrapperFor: titleBar) click.
+	
+	(self assertReading: groupChat description in: (self subject findByClass: TCUChatInfoPage)).
```
Comment (phkeese) [inline packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testDoesDisplayGroupDescription.st:16]: why the parenthesis around the line?

File: packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testPrivateChatHasNoDescription.st [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+		whileFalse: [6 seconds wait.].
+	groupChat := (self core chats detect: [:chat | chat isPrivate]).
+	
+	(self subject findByCriteria: [ :morph | (morph respondsTo: #chatID) and: [morph chatID = groupChat id]]) click.
+	titleBar := (self subject findByClass: TCUChatWindow) morphs collect: [:chatWindow | chatWindow titleBar].
+	(self subject createWrapperFor: titleBar) click.
+
+	chatInfoPage := (self subject findByClass: TCUChatInfoPage) morphs anyOne.
+	self forceSaveScreenshotOf: chatInfoPage as: 'testPrivateChatHasNoDescription' in: self class defaultTestResultPath.
+	self assert: 134349208 equals: (self computeHashOf: (self takeScreenshotOf: chatInfoPage)).
```
Comment (phkeese) [inline packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testPrivateChatHasNoDescription.st:17]: maybe put the constant into an accessor

File: packages/TelegramClient-UI.package/TCUChatWindow.class/instance/addHiddenInfoPage.st [hunk 12]
```diff
@@ -2,8 +2,7 @@ drawing
 addHiddenInfoPage
 
 	self
-		infoPage: ((TCUChatInfoPage newBounds: self bounds)
-			yourself);
+		infoPage: (TCUChatInfoPage newFor: self selectedChat withBounds: self bounds);
 		addMorphFront: self infoPage.
-	
+
```
Comment (phkeese) [inline packages/TelegramClient-UI.package/TCUChatWindow.class/instance/addHiddenInfoPage.st:7]: please remove

**Review by phkeese — CHANGES_REQUESTED** [review]
Some formatting looks weird and in a few places you access the callbacks too directly

File: packages/TelegramClient-Core.package/TCCChat.class/instance/isNotPrivate.st (hunk 1 above)
Comment (jb3rndt) [reply packages/TelegramClient-Core.package/TCCChat.class/instance/isNotPrivate.st:4]: We added the `isGroup` so this was not used anymore. I guess we can add the negation again later if we need it. On the other hand, `isGroup` is already the counter part of this, but that might change in the future.

File: packages/TelegramClient-Core.package/TCCGroup.class/instance/isGroup.st (hunk 2 above)
Comment (jb3rndt) [reply packages/TelegramClient-Core.package/TCCGroup.class/instance/isGroup.st:4]: Hm I dont know; We found that `isGroup` is more descriptive and better understandable than `isNotPrivate`.

File: packages/TelegramClient-Core.package/TCCChat.class/class/newFromChatEvent..st (hunk 4 above)
Comment (jb3rndt) [reply packages/TelegramClient-Core.package/TCCChat.class/class/newFromChatEvent..st:8]: Do you mean something like this?
```
newFromChatEvent: anEvent

	| type |
	type := ((anEvent at: 'chat') at: 'type') at: '@type'.
	
	((type = 'chatTypeSupergroup') or: (type = 'chatTypeBasicGroup'))
		ifTrue: [^ TCCGroup new initializeFromChatEvent: anEvent.]
		ifFalse: [^ self new initializeFromChatEvent: anEvent.].
```
We could override `TCCGroup >> newFromChatEvent:` to use that directly but we would have to write a second constructor for the `TCCChat` to do that there too, which seemed a bit weird because we would then have the `newFromChatEvent:` just for the decision whether its a group or not.
What do you suggest would be the best way?

File: packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testPrivateChatHasNoDescription.st (hunk 11 above)
Comment (jb3rndt) [reply packages/TelegramClient-Tests.package/TCTAcceptanceTests.class/instance/testPrivateChatHasNoDescription.st]: I know magic numbers are a thing but these hashes are so unique and a core part of the test that I'm not sure whether we have a benefit from writing an accessor for each hash

**Review by phkeese — APPROVED** [review]
